libqtile/widget/check_updates.py

```diff
@@ -97,6 +97,9 @@ def poll(self):
 
     def button_press(self, x, y, button):
         # type: (int, int, int) -> None
-        base.ThreadedPollText.button_press(self, x, y, button)
+        name = 'Button{0}'.format(button)
+        if name in self.mouse_callbacks:
+            self.mouse_callbacks[name](self.qtile)
+            return
         if button == 1 and self.execute is not None:
```

Comment:
Sorry, I was thinking more that we hoist dispatch of all the other calculations into a button press, rather than copy paste this into every widget.

So for this one, we'd register a callback for Button1 that is the body of this function.

It's also a bit awkward that we pass the qtile object to the buttons, since for the most part I assume they'll want to operate on themselves. I wonder if we can change it to pass the button?

Reply:
For your first comment - I agree that that would be much better, but figured this provide the functionality for all widgets in the meantime while they each get ported over time. Understandable if you'd rather not add this in this way (it's not particularly pretty code).

For your second comment - do you mean just the button string i.e. `Button1`, or the widget's self?

Reply:
I was thinking the widget's self reference. Yeah, I'd rather not merge this, I was thinking a port some/all the widgets would be better.

Reply:
No worries at all, I'll add a couple of these widgets to my to-do list!

Reply:
#1713